modules/src/ics04_channel/handler/chan_open_init.rs

```diff
@@ -0,0 +1,184 @@
+//! Protocol logic specific to ICS3 messages of type `MsgChannelOpenInit`.
+
+use crate::handler::{HandlerOutput, HandlerResult};
+use crate::ics04_channel::channel::{ChannelEnd, State};
+use crate::ics04_channel::context::ChannelReader;
+use crate::ics04_channel::error::{Error, Kind};
+use crate::ics04_channel::handler::ChannelEvent::ChanOpenInit;
+use crate::ics04_channel::handler::ChannelResult;
+use crate::ics04_channel::msgs::chan_open_init::MsgChannelOpenInit;
+
+pub(crate) fn process(
+    ctx: &dyn ChannelReader,
+    msg: MsgChannelOpenInit,
+) -> HandlerResult<ChannelResult, Error> {
+    let mut output = HandlerOutput::builder();
+
+    let pc_id = (msg.port_id().clone(), msg.channel_id().clone());
+
+    if ctx.channel_end(&pc_id).is_some() {
+        return Err(Kind::ChannelExistsAlready(msg.channel_id().clone()).into());
+    }
+    //TODO what about the port capabilities  ?
```

Comment:
```suggestion
    // TODO: What about the port capabilities?
```

Reply:
I didn't see any capabilities. Do we ignore them for now/all together ?

Reply:
Added a place holder for capabilities.

Reply:
Yep, we're ignoring capabilities for now.